docs/xocto/ranges.md

```diff
@@ -0,0 +1,26 @@
+# Ranges
+
+## A utility for working with ranges
+
+The basic building block of the module is the `Range` class.
+
+## Usage
+
+A few examples:
+
+```python
+from xocto.ranges import Range, RangeBoundaries
+
+>>> Range(0, 2, boundaries=RangeBoundaries.EXCLUSIVE_INCLUSIVE)
+<Range: (0,2]>
+>>> Range(0, 2, boundaries="[]")
+<Range: [0,2]>
+>>> sorted([Range(1, 4), Range(0, 5)])
+[<Range: [0,5)>, <Range: [1,4)>]
+>>> sorted([Range(1, 2), Range(None, 2)])
+[<Range: [None,2)>, <Range: [1,2)>]
+>>> sorted([Range(3, 5), Range(3, 4)])
+[<Range: [3,4)>, <Range: [4,5)>]
+```
```

Comment:
Suggestion, nonblocking: demo a wider range of cooler stuff:

```suggestion
from xocto.ranges import Range, RangeBoundaries

>>> Range(0, 2, boundaries=RangeBoundaries.EXCLUSIVE_INCLUSIVE)
<Range: (0,2]>
>>> 0 in Range(0, 2)
True
>>> 2 in Range(0, 2)
False
>>> date(2020, 1, 1) in Range(date(2020, 1, 2), date(2020, 1, 5))
False
>>> sorted([Range(1, 4), Range(0, 5)])
[<Range: [0,5)>, <Range: [1,4)>]
```

Reply:
Can you submit this in a future PR?